src/odemis/gui/cont/tabs/fastem_setup_tab.py

```diff
@@ -275,39 +266,35 @@ def _on_optical_autofocus_done(self, future, _=None):
                 "Optical Autofocus calibration failed with exception: %s.", ex
             )
         finally:
-            self._update_optical_autofocus_controls()
-            self.calibration_controller.calibration_panel.Enable(True)
+            self._update_button_controls(self.btn_optical_autofocus)
+            # Resume SettingEntry related control updates of the stream
+            self.sem_stream_cont.resume()
+            if self.stream_should_update:
+                self.tab_data.semStream.should_update.value = True
 
-    @wxlimit_invocation(0.1)  # max 10Hz; called in main GUI thread
-    def _update_optical_autofocus_controls(self, button_state=True):
-        """
-        Update the optical autofocus button controls to allow cancelling or a re-run.
-        :param button_state: (bool) Enabled or disable button depending on state. Default is enabled.
-        """
-        self.btn_optical_autofocus.Enable(button_state)  # enable/disable button
-
-        if self.tab_data.is_calibrating.value:
-            self.btn_optical_autofocus.SetLabel(
-                "Cancel"
-            )  # indicate cancelling is possible
-        else:
-            self.btn_optical_autofocus.SetLabel(
-                "Run"
-            )  # change button label back to ready for calibration
-
-        self.sem_stream_cont.stream_panel.Layout()
-        self.sem_stream_cont.stream_panel.Refresh()
-
-    @call_in_wx_main
     def _on_btn_sem_autofocus(self, _):
         # Disable other calibration buttons
         # TODO also disable btn_autostigmation once autostigmation is working
+        if self.tab_data.is_calibrating.value:
+            fastem._executor.cancel()
```

Comment:
where is `fastem` defined?

Reply:
Here: https://github.com/delmic/odemis/blob/dd5d4cf75cdcf5fddd188f19127536d4cbdd8525/src/odemis/acq/fastem.py#L54